src/libs/elektra/keyset.c

```diff
@@ -357,6 +405,14 @@ int ksDel (KeySet * ks)
 	if (!ks) return -1;
 
 	rc = ksClose (ks);
+
+#ifdef ELEKTRA_ENABLE_OPTIMIZATIONS
```

Comment:
Doesn't it make more sense to have this code in ksClose?

Reply:
The `Opmphm` struct is designed to be reused, so the only place to completely free it is `ksDel (....)`.
`ksClose (...)` is also invoked from `ksClear (...)`, I think deleting the `Opmphm` struct on every `ksClear (...)` is not necessary.

Reply:
I added a invalidation in the `ksClose`.